src/tracks/ui/BrushHandle.cpp

```diff
@@ -224,17 +224,20 @@ bool BrushHandle::Escape(AudacityProject *project)
 }
 
 // Add or remove data accroding to the ctrl key press
-void BrushHandle::HandleTimeFreqData(long long ll_sc, wxInt64 freq) {
+void BrushHandle::HandleHopBinData(int hopNum, int freqBinNum) {
    // Ignore the mouse dragging outside valid area
    // We should also check for the available freq. range that is visible to user
-   if(ll_sc < mSampleCountLowerBound || ll_sc > mSampleCountUpperBound ||
+   long long sampleCount = hopNum * mpSpectralData->GetHopSize();
+   wxInt64 freq = freqBinNum * mpSpectralData->GetSR() / mpSpectralData->GetWindowSize();
+
+   if(sampleCount < mSampleCountLowerBound || sampleCount > mSampleCountUpperBound ||
       freq < mFreqLowerBound || freq > mFreqUpperBound)
```

Comment:
I checked again ho mFreqLowerBound and mFreqUpperBound are set up when you construct the brush handle.  You still exclude the lowest and hightest frequency bins from editing.  Is that still needed after better bounds checking and removing the +/- 1 hack?

Reply:
Yes, since dragging out of the track (to non-visible frequency) will generate wrong result for connecting coordinates.